easybuild/easyblocks/e/easybuildmeta.py

```diff
@@ -65,6 +65,9 @@ def __init__(self, *args, **kwargs):
             self.easybuild_pkgs.extend(['vsc-base', 'vsc-install'])
             # consider setuptools first, in case it is listed as a sources
             self.easybuild_pkgs.insert(0, 'setuptools')
+        elif LooseVersion(self.version) >= LooseVersion('5.0'):
+            # use easybuild-base for easybuild to avoid matching all easybuild-* directories during install
+            self.easybuild_pkgs.append('easybuild-base')
```

Comment:
what if we guard this to also check that the `easybuild-{version}.tar.gz` package is in the sources? Then in theory this could be merged without https://github.com/easybuilders/easybuild-easyconfigs/pull/22774 for now?
```suggestion
        elif LooseVersion(self.version) >= LooseVersion('5.0') and any(f'easybuild-{self.version}' in source['filename'] for source in self.cfg['sources']):
            # use easybuild-base for easybuild to avoid matching all easybuild-* directories during install
            self.easybuild_pkgs.append('easybuild-base')
```

Reply:
The issue I am expecting is when the new source appears in the easyconfig it will break because someone mayl be using an old easybuild that does not have whatever protection we add in here.

Reply:
indeed; with the addition of the suggestion we could
* merge this as is
* then merge https://github.com/easybuilders/easybuild-easyconfigs/pull/22774 (or equivalent) at some future date (EB 6.0?) -- once this code has been in a few releases it would be less of an issue

that would mean manually generating the egg file would be a temporary workaround